app/lib/dartdoc/backend.dart

```diff
@@ -375,9 +401,29 @@ class DartdocBackend {
   }
 
   Future<void> _deleteAll(DartdocEntry entry, {int concurrency}) async {
+    await withRetryTransaction(_db, (tx) async {
+      final r = await tx.lookupValue<DartdocRecord>(
+        _db.emptyKey.append(DartdocRecord, id: entry.uuid),
+        orElse: () => null,
+      );
+      if (r != null) {
+        r.status = DartdocRecordStatus.deleting;
```

Comment:
Why do we need to change it to `deleting`?

This should have `'runtimeVersion < ', shared_versions.gcBeforeRuntimeVersion`, I don't think there any chance we'll try to use it, right?

So I don't think we need to write that we're currently deleting it.

It might make sense if we wanted to avoid concurrent deletion, but then we need something that removes this lock after a timeout... and if we use neat-periodic-task to schedule this, then there is rarely two cron jobs running at the same time.. Moreover, I'm not sure it's the end of the world if we accidentally did some concurrent deletion.

Reply:
The same runtimeVersion could create a second content. When that happens, we can start to delete the older version, as we won't be serving it anyway. (This is one of the pending unmigrated periodic task.)

I don't think it is crucial to expire these deletes, as eventually it will be caught by the runtimeVersion-based cleanups.